zerver/lib/upload.py

```diff
@@ -120,40 +119,20 @@ def ensure_medium_avatar_image(self, email):
 
 ### S3
 
-def get_bucket(conn, bucket_name):
-    # type: (S3Connection, Text) -> Bucket
-    # Calling get_bucket() with validate=True can apparently lead
-    # to expensive S3 bills:
-    #    http://www.appneta.com/blog/s3-list-get-bucket-default/
-    # The benefits of validation aren't completely clear to us, and
-    # we want to save on our bills, so we set the validate flag to False.
-    # (We think setting validate to True would cause us to fail faster
-    #  in situations where buckets don't exist, but that shouldn't be
-    #  an issue for us.)
-    bucket = conn.get_bucket(bucket_name, validate=False)
-    return bucket
```

Comment:
http://boto3.readthedocs.io/en/latest/guide/migrations3.html#accessing-a-bucket
boto3 doesn't automatically validate whether a bucket exists so in normal calls validate will be false and for validation we have to check manually as mentioned in above link.